test/utils.mjs

```diff
@@ -221,6 +221,15 @@ export class Web3Client {
       if (timeout === 0) throw new Error('Timeout in waitForEvent');
     }
 
+    if (expectedEvents[0] === 'blockProposed') {
```

Comment:
I would rewrite `waitForEvent` instead of doing this explicitly for `blockProposed` insde. Instead of always looking at eventLogs[0], I'd search in all the array for the event and do the push or shift operations based on the specific `expectedEvents` passed.

Reply:
Also in the `start-nightfall` the `nightfall_3_mongodb-adversary-db` volume should be removed if present. If it's not removed then the test only works the first time.